misc/chrome_extension/panel.js

```diff
@@ -199,7 +208,7 @@
 			getStorageValues(function (results) {
 				results.forEach(function (element) {
```

Comment:
I missed that before; try to limit how often you change the DOM .
Instead of changing the innerHTML for each option, you should build the HTML first, then change innerHTML once. 

Something like: 
```
let select = document.getElementById('storage');
let optionsHtml = ['<option selected disabled>Select a file to work on</option><option disabled>------------------------------</option>'];

  			getStorageValues(function (results) {
results.forEach(element => {
	optionsHtml.push(`<option value="${element}">${element}</option>`);
});
optionsHtml.push('<option value="__create">Create new file...</option>');

select.innerHTML = optionsHtml.join('');
```